packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -209,21 +213,28 @@ export default class ScriptTransformer {
             // files outside `cwd` will not be instrumented
             cwd: this._config.rootDir,
             exclude: [],
+            // Needed for correct coverage as soon as we start storing a source map of the instrumented code
+            inputSourceMap: input.map,
             useInlineSourceMaps: false,
           },
         ],
       ],
+      /**
+       * Necessary to be able to map back to original source from the instrumented code
+       * the inline map is needed for debugging functionality
+       * and exposing it as a separate file is needed for e.g. mapping stack traces
+       * convenient to use 'both' here and avoid extracting the source map
+       *
+       * Prior behavior of emitting no map when we can't map back to original source is preserved
+       */
+      sourceMaps: canMapToInput ? 'both' : false,
     });
 
-    if (result) {
-      const {code} = result;
```

Comment:
I did this juggle at some point as the types were confused. TS seems.happy now though, so I'm happy as well 🙂